### PR TITLE
Camera: Arm inertia code cleanup

### PR DESCRIPTION
**Pull request by kilbith**
It no longer depends on the frame time + code shortening.

### CODE REVIEW

File: src/camera.cpp [hunk 1]
```diff
@@ -269,28 +258,21 @@ void Camera::addArmInertia(f32 player_yaw, f32 frametime)
 		    following a vector, with a smooth deceleration factor.
 		*/
 
-		f32 acc_X = (m_cam_vel_old.X * (1.0f + (1.0f - m_arm_dir.X))) /
-			(20.0f / std::fabs(55.0f - m_wieldmesh_offset.X));
-		f32 acc_Y = (m_cam_vel_old.Y * (1.0f + (1.0f - m_arm_dir.Y))) /
-			(15.0f / std::fabs(-35.0f - m_wieldmesh_offset.Y));
+		f32 dec_X = 0.12f * (m_cam_vel_old.X * (1.0f +
+			(1.0f - m_arm_dir.X))) / (20.0f / gap_X);
```
Comment (nerzhul) [inline src/camera.cpp]: you can prevent a zéro division here by changing maths, it also permit to have a less CPU cost function (division costs far more than multiplication for CPU)

Replace
```
/ (20.0f / gap_X)
```
With:
```
* (gap_X / 20.0f)
```

File: src/camera.cpp [hunk 2]
```diff
@@ -269,28 +258,21 @@ void Camera::addArmInertia(f32 player_yaw, f32 frametime)
...
 
-		if (std::fabs(55.0f - m_wieldmesh_offset.X) < 0.1f)
-			m_cam_vel_old.X = 0.0f;
+		f32 dec_Y = 0.06f * (m_cam_vel_old.Y * (1.0f +
+			(1.0f - m_arm_dir.Y))) / (15.0f / gap_Y);
```
Comment (nerzhul) [inline src/camera.cpp]: same here

File: src/camera.cpp [hunk 3]
```diff
@@ -211,10 +211,12 @@ static inline v2f dir(const v2f &pos_dist)
...
 }
 
-void Camera::addArmInertia(f32 player_yaw, f32 frametime)
+void Camera::addArmInertia(f32 player_yaw)
 {
-	m_cam_vel.X = std::fabs((m_last_cam_pos.X - player_yaw) / frametime) * 0.01f;
-	m_cam_vel.Y = std::fabs((m_last_cam_pos.Y - m_camera_direction.Y) / frametime);
+	m_cam_vel.X = std::fabs((m_last_cam_pos.X - player_yaw) / 0.016f) * 0.01f;
+	m_cam_vel.Y = std::fabs((m_last_cam_pos.Y - m_camera_direction.Y) / 0.016f);
+	f32 gap_X = std::fabs(55.0f - m_wieldmesh_offset.X);
```
Comment (nerzhul) [inline src/camera.cpp]: maybe add macros for 55.0f and -35.0f to define textualy what it is (it's used in two places)

File: src/camera.cpp (hunk 1 above)
Comment (sfan5) [inline src/camera.cpp]: > it also permit to have a less CPU cost function (division costs far more than multiplication for CPU)

While you're correct, I am fairly sure that gcc will perform such optimizations on its own if applicable.

**Pull request merged**